src/_pytest/python.py

```diff
@@ -1326,10 +1326,12 @@ def parametrize(
         # more than once) then we accumulate those calls generating the cartesian product
         # of all calls.
         newcalls = []
-        for callspec in self._calls or [CallSpec2()]:
+        for callspec_index, callspec in enumerate(self._calls or [CallSpec2()]):
             for param_index, (param_id, param_set) in enumerate(
                 zip(ids, parametersets)
             ):
+                if name2pseudofixturedef:
+                    param_index = callspec_index + param_index
```

Comment:
Shouldn't we have

```suggestion
                    param_index = callspec_index * len(parametersets) + param_index
```
? Because currently some callspecs would have arguments with the same param index but different param values. For example, `test1[1-b]` and `test1[2-a]` whose `arg2` param index is 1. However this doesn't show up in this test. In

```python
def test_multi_parametrize_reordering(self, pytester: Pytester) -> None:
        pytester.makepyfile(
            """
            import pytest

            @pytest.mark.parametrize("arg2", ['a', 'b', 'c'], scope='class')
            @pytest.mark.parametrize("arg1", [1, 2], scope='class')
            class TestClass:
                def test1(self, arg1, arg2):
                    pass

                def test2(self, arg1, arg2):
                    pass
        """
        )
        result = pytester.runpytest("--collect-only")
        result.stdout.re_match_lines(
            [
                r"      <Function test1\[1-a\]>",
                r"      <Function test2\[1-a\]>",
                r"      <Function test1\[1-b\]>",
                r"      <Function test2\[1-b\]>",
                r"      <Function test1\[1-c\]>",
                r"      <Function test2\[1-c\]>",
                r"      <Function test1\[2-a\]>",
                r"      <Function test2\[2-a\]>",
                r"      <Function test1\[2-b\]>",
                r"      <Function test2\[2-b\]>",
                r"      <Function test1\[2-c\]>",
                r"      <Function test2\[2-c\]>",
            ]
        )
```
it raises error. The test output would be:
```
<Dir test_multi_parametrize_reordering0>
  <Module test_multi_parametrize_reordering.py>
    <Class TestClass>
      <Function test1[1-a]>
      <Function test2[1-a]>
      <Function test1[1-b]>
      <Function test2[1-b]>
      <Function test1[2-a]>
      <Function test2[2-a]>
      <Function test1[2-b]>
      <Function test2[2-b]>
      <Function test1[1-c]>
      <Function test2[1-c]>
      <Function test1[2-c]>
      <Function test2[2-c]>
```

Reply:
I would also suggest removing the "if name2pseudofixturedef" check so that sorting parameterized tests through a marker and through a fixture works the same way.
But then it is necessary to correct the tests test_module_parametrized_ordering and test_dynamic_parametrized_ordering along the way